index.bs

```diff
@@ -371,13 +389,113 @@ model]]:
 
 1. Run [=record a user activation=] given <var ignore>document</var>.
 
+<h4 id="bounce-tracking-mitigation-stateful-bounce-detection">Stateful Bounce
+Detection</h4>
+
+<div algorithm>
+
+To <dfn>process a network cookie access for bounce tracking mitigations</dfn>
+given a [=request=] |request|, perform the following steps:
+
+1. Let |origin| be |request|'s [=request/origin=].
+1. If |origin| is an [=opaque origin=], then abort these steps.
+1. If |request|'s [=request/destination=] is "`document`", then:
+    1. Let |topLevelTraversable| be |request|'s [=request/client=]'s
+        [=environment/target browsing context=]'s
+        [=bc-traversable|top-level traversable=].
+    1. [=Assert=]: |topLevelTraversable|'s
+        [=top-level traversable/bounce tracking record=] is not null.
+    1. [=set/Append=] |origin|'s [=host=] to |topLevelTraversable|'s
+        [=top-level traversable/bounce tracking record=]'s
+        [=bounce tracking record/storage access set=].
+
+Issue: TODO: Handle subresource requests and iframes for client-side redirects.
+
+</div>
+
+<h5 id="bounce-tracking-mitigations-network-cookie-write-monkey-patch">Network
+Cookie Write Monkey Patch</h5>
+
+Insert the following steps in the <a spec="fetch">HTTP-network fetch</a>
+algorithm after step 15 where cookies are parsed and saved in the cookie store:
+
+1. If cookies were stored in the cookie store in the previous step, then
+    run [=process a network cookie access for bounce tracking mitigations=]
+    given <var ignore>request</var>.
+
+<h5 id="bounce-tracking-mitigations-network-cookie-read-monkey-patch">Network
```

Comment:
So it turns out we are not running this algorithm on cookie read right now.  We only treat cookie writes as stateful at the moment.  You could turn this into a Note that treating cookie reads as stateful would be a reasonable future change.

Reply:
Done.